pkg/scheduler/factory/factory.go

```diff
@@ -849,15 +843,17 @@ func (c *configFactory) addNodeToCache(obj interface{}) {
 		return
 	}
 
-	if err := c.schedulerCache.AddNode(node); err != nil {
-		glog.Errorf("scheduler cache AddNode failed: %v", err)
-	}
-
+	// NOTE: Because the scheduler uses equivalence cache for nodes, we need
+	// to create it before adding node into scheduler cache.
 	if c.enableEquivalenceClassCache {
 		// GetNodeCache() will lazily create NodeCache for given node if it does not exist.
 		c.equivalencePodCache.GetNodeCache(node.GetName())
```

Comment:
Since we will init NodeCache here, it seems we don't need to call `g.equivalenceCache.GetNodeCache(nodeName)` in generic_scheduler?

Instead, we can only call sth like `g.equivalenceCache.LoadNodeCache()` there, only read operation. WDYT?